pennylane/qchem/vibrational/bosonic.py

```diff
@@ -0,0 +1,639 @@
+        r"""String representation of a BoseWord."""
+        return f"{self.to_string()}"
+
+    def __repr__(self):
+        r"""Terminal representation of a BoseWord"""
+        return f"BoseWord({self.sorted_dic})"
+
+    def __add__(self, other):
+        """Add a BoseSentence, BoseWord or constant to a BoseWord. Converts both
+        elements into BoseSentences, and uses the BoseSentence __add__
+        method"""
+
+        self_bs = BoseSentence({self: 1.0})
+
+        if isinstance(other, BoseSentence):
+            return self_bs + other
+
+        if isinstance(other, BoseWord):
+            return self_bs + BoseSentence({other: 1.0})
+
+        if not isinstance(other, TensorLike):
+            raise TypeError(f"Cannot add {type(other)} to a BoseWord.")
+
+        if qml.math.size(other) > 1:
+            raise ValueError(
+                f"Arithmetic Bose operations can only accept an array of length 1, "
+                f"but received {other} of length {len(other)}"
+            )
+
+        return self_bs + BoseSentence({BoseWord({}): other})
+
+    def __radd__(self, other):
+        """Add a BoseWord to a constant, i.e. `2 + BoseWord({...})`"""
+        return self.__add__(other)
+
+    def __sub__(self, other):
+        """Subtract a BoseSentence, BoseWord or constant from a BoseWord. Converts both
+        elements into BoseSentences (with negative coefficient for `other`), and
+        uses the BoseSentence __add__  method"""
+
+        self_bs = BoseSentence({self: 1.0})
+
+        if isinstance(other, BoseWord):
+            return self_bs + BoseSentence({other: -1.0})
+
+        if isinstance(other, BoseSentence):
+            other_bs = BoseSentence(dict(zip(other.keys(), [-v for v in other.values()])))
+            return self_bs + other_bs
+
+        if not isinstance(other, TensorLike):
+            raise TypeError(f"Cannot subtract {type(other)} from a BoseWord.")
+
+        if qml.math.size(other) > 1:
+            raise ValueError(
+                f"Arithmetic Bose operations can only accept an array of length 1, "
+                f"but received {other} of length {len(other)}"
+            )
+
+        return self_bs + BoseSentence({BoseWord({}): -1 * other})  # -constant * I
+
+    def __rsub__(self, other):
+        """Subtract a BoseWord to a constant, i.e. `2 - BoseWord({...})`"""
+        if not isinstance(other, TensorLike):
+            raise TypeError(f"Cannot subtract a BoseWord from {type(other)}.")
+
+        if qml.math.size(other) > 1:
+            raise ValueError(
+                f"Arithmetic Bose operations can only accept an array of length 1, "
+                f"but received {other} of length {len(other)}"
+            )
+        self_bs = BoseSentence({self: -1.0})
+        other_bs = BoseSentence({BoseWord({}): other})
+        return self_bs + other_bs
+
+    def __mul__(self, other):
+        r"""Multiply a BoseWord with another BoseWord, a BoseSentence, or a constant.
```

Comment:
Does this take care of hardcode boson thing if the provided bosewords have is_hardcore True?

Reply:
No, I don't think so. But suppose you have one bw that is hard core and another that isn't. How do we define the behaviour for when they're multiplied?